Update Web Bluetooth article

beaufortfrancois opened the pull request:
Here's an update to the Web Bluetooth article.
Live preview at https://updatewebbluetootharticle-dot-weasel-dot-web-central.appspot.com/web/updates/2015/07/interact-with-ble-devices-on-the-web
R=@scheib @jyasskin @g-ortuno

g-ortuno commented on src/content/en/updates/posts/2015/07/interact-with-ble-devices-on-the-web.markdown:
```diff
@@ -50,7 +50,8 @@ enable the highlighted flag, restart Chrome and you should be able to
 [read](#read-a-bluetooth-characteristic)/[write](#write-to-a-bluetooth-characteristic)
-Bluetooth characteristics and [receive GATT Notifications](#receive-gatt-notifications).
+Bluetooth characteristics, [receive GATT Notifications](#receive-gatt-notifications) and know when [Bluetooth device gets
+disconnected](#get-disconnected-from-a-bluetooth-device).
```
Is this really that special that needs to be mentioned?

beaufortfrancois replied:
I've only added it there to be able to jump directly to the section.

g-ortuno commented on src/content/en/updates/posts/2015/07/interact-with-ble-devices-on-the-web.markdown:
```diff
@@ -231,12 +253,15 @@ navigator.bluetooth.requestDevice({ filters: [{ services: ['heart_rate'] }] })
 .catch(error => { console.log(error); });
 {% endhighlight %}
 
+Using a `characteristicvaluechanged` event listener to know when a characteristic
```
Wait, is this actually implemented? If so, that's a bug: the spec doesn't say anything about an event being dispatched after writing a value.

beaufortfrancois replied:
Is https://bugs.chromium.org/p/chromium/issues/detail?id=592659 wrong then?
I'll also update our web Bluetooth sample as well then.

beaufortfrancois replied:
Done.
Fixed in https://github.com/GoogleChrome/samples/pull/374/commits/250f64a02af4f5461e7f6e407aa7d3fd37d67343 as well.

g-ortuno commented on src/content/en/updates/posts/2015/07/interact-with-ble-devices-on-the-web.markdown:
```diff
@@ -249,48 +274,82 @@ navigator.bluetooth.requestDevice({ filters: [{ services: ['heart_rate'] }] })
+  // Attempts to connect to remote GATT Server.
+  return device.gatt.connect();
+})
+.then(server => {...})
```
optional nit: Moving this to the next line and putting the "..." in a comment would mean that anyone could copy and paste this code and run it without any modifications. WDYT?

beaufortfrancois replied:
Good idea!

jpmedley commented on src/content/en/updates/posts/2015/07/interact-with-ble-devices-on-the-web.markdown:
```diff
@@ -50,7 +50,8 @@ enable the highlighted flag, restart Chrome and you should be able to
 nearby Bluetooth devices,
 [read](#read-a-bluetooth-characteristic)/[write](#write-to-a-bluetooth-characteristic)
-Bluetooth characteristics and [receive GATT Notifications](#receive-gatt-notifications).
+Bluetooth characteristics, [receive GATT Notifications](#receive-gatt-notifications) and know when [Bluetooth device gets
```
'and know when Bluetooth device' > 'and know when a Bluetooth device'. Please fix.

beaufortfrancois replied:
Done.

jpmedley commented on src/content/en/updates/posts/2015/07/interact-with-ble-devices-on-the-web.markdown:
```diff
@@ -249,48 +274,82 @@ navigator.bluetooth.requestDevice({ filters: [{ services: ['heart_rate'] }] })
+### Get disconnected from a Bluetooth Device
+
+In order to provide a better user experience, you may want to show a warning
+message if the `BluetoothDevice` gets disconnected to invite user to reconnect.
```
'user' > 'the user'.

beaufortfrancois replied:
Done.

jpmedley commented on src/content/en/updates/posts/2015/07/interact-with-ble-devices-on-the-web.markdown:
```diff
@@ -249,48 +274,82 @@ navigator.bluetooth.requestDevice({ filters: [{ services: ['heart_rate'] }] })
+the Bluetooth device. Note that it will NOT stop bluetooth device communication
+if another app is already communicating with the Bluetooth device. Check out the [Device Disconnect Sample](https://googlechrome.github.io/samples/web-bluetooth/device-disconnect.html) and the [Automatic Reconnect Sample](https://googlechrome.github.io/samples/web-bluetooth/automatic-reconnect.html) to dive deeper.
+
 ## Samples, Demos & Codelabs
```
Please use and instead of '&'.

beaufortfrancois replied:
Done.

jpmedley commented on src/content/en/updates/posts/2015/07/interact-with-ble-devices-on-the-web.markdown:
```diff
@@ -310,29 +369,35 @@ I would recommend you check out the official [Bluetooth debug page](https://site
+status](https://github.com/WebBluetoothCG/web-bluetooth/blob/gh-pages/implementation-status.md)
+first to know which parts of the Web Bluetooth API are currently being implemented.
+
+As it is still not complete yet, here's a sneak peek of what to expect in the
```
'As it is still not complete yet' > 'Though it's still incomplete'. Or something similar.

beaufortfrancois replied:
Done.

jpmedley commented:
I found a few things that need to be fixed before we can merge this.

beaufortfrancois commented:
Thank you @jpmedley! I'll fix those.

beaufortfrancois commented on src/content/en/updates/posts/2015/07/interact-with-ble-devices-on-the-web.markdown:
```diff
@@ -249,48 +274,82 @@ navigator.bluetooth.requestDevice({ filters: [{ services: ['heart_rate'] }] })
 
+### Get disconnected from a Bluetooth Device
+
+In order to provide a better user experience, you may want to show a warning
```
Done.

beaufortfrancois commented:
@g-ortuno @jpmedley I've addressed your feedback at https://github.com/google/WebFundamentals/pull/3025/commits/2b99788d48e8d6801e34dd176e88bb2be7ec20ca.
If that looks good to you, I'll ask @petele to merge and push it live.

g-ortuno commented on src/content/en/updates/posts/2015/07/interact-with-ble-devices-on-the-web.markdown:
```diff
@@ -249,48 +271,82 @@ navigator.bluetooth.requestDevice({ filters: [{ services: ['heart_rate'] }] })
+}
+{% endhighlight %}
+
+You can also call `device.gatt.disconnect()` to disconnect your web app from
```
Not sure if you have to include this but note that calling `disconnect()` would also trigger the event.

beaufortfrancois replied:
Good catch! Will add it.

g-ortuno commented:
bluetooth code lgtm bar one nit

beaufortfrancois commented:
@g-ortuno's feedback is addressed.
Please @petele @PaulKinlan, can we merge it and push this live?

petele commented:
merged - will push live later today or early tomorrow.

beaufortfrancois commented:
It is now live at https://developers.google.com/web/updates/2015/07/interact-with-ble-devices-on-the-web Thank you @petele!
